### PR TITLE
[addons][viz] code cleanup / improvements / API update

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This cleans up the visualization add-on system by take of code from my sandbox project. Also for here relates not direct to this project and can be brought in independent.

Main change is to remove the fourier transformation inside Kodi and do only by add-ons (was only one still there, where used them from Kodi).
The other is to cleanup the API and have better in overview.

All related changes are separated in own commits with short descriptions about.

This should be the last API change added from me for Nexus and want to make all others in coming Kodi O*.

## How has this been tested?
With changed add-ons.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/guilib/GUIVisualisationControl.h [hunk 1]
```diff
@@ -95,5 +95,5 @@ class CGUIVisualisationControl : public CGUIControl, public IAudioCallback
   std::string m_presetsPath; /*!< To add-on sended preset path */
   std::string m_profilePath; /*!< To add-on sended profile path */
 
-  ADDON::CVisualization* m_instance{nullptr};
+  KODI::ADDONS::CVisualization* m_instance{nullptr};
```
Comment (garbear) [inline xbmc/guilib/GUIVisualisationControl.h:98]: `#include "addons/Visualization.h"` can be moved to the cpp file and this class forward-declared.

**Review by garbear — APPROVED** [review]
Nice cleanup. I see how the previous PRs progressed to this one.

Only comment would be to clang-format the entire xbmc/addons (you've earned this folder) and be done with it.

File: xbmc/guilib/GUIVisualisationControl.h (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/guilib/GUIVisualisationControl.h:98]: Thanks for the note, will change it.
Also you brought me to an idea to change it to `std::unique_ptr` to confirm destruct in all cases.

**Comment by fuzzard** [comment]
Jenkins build this please

**Pull request merged**